10-algorithms.Rmd

```diff
@@ -33,7 +33,8 @@ That is the topic of section \@ref(functions).
 <!-- nor entirely focused on implementations via a GUI or CLI in a particular sofware package (as the first part of this book is, with its focus on implementations in various R packages). -->
 <!-- The focus of this chapter is on understanding, using reproducible code and clear explanation. -->
 The example of finding the centroid of a polygon will be used to tie these concepts together.
-Of course, you already know how to do that with `st_centroid()` having read Chapter \@ref(geometric-operations).
+Of course, you already know how to do that with `st_centroid()` having read Chapter \@ref(geometric-operations). 
+<!-- JM: I think the quote refers to something else, e.g., humans can easily recognize faces or can easily take a step, something which is very hard for a computer/roboter. centroid computation might not be the most complex procedure in the world, still, when asking your students how to do so you will find that they will not find it as easy as recognizing a friend -->
```

Comment:
I think this *is* what the quote is referring to. Here's the quote in extended context:

> Computer software is generally designed to undertake particular tasks or
to solve problems. Geographic Information Systems (GIS) are no excep-
tion, and of course in this case the tasks or problems involve spatial
data. One of the most intriguing things about spatial data problems is
that things which appear to be trivially easy to a human being can be
surprisingly difficult on a computer. This is partly due to the complex
nature of spatial data, and partly to the way that computers tackle
problems.

Happy to consider suggestions to change the prose, but I think what is written is what we mean: devising algorithms is hard.

Reply:
yeah, ok, then we should mention, that it is relatively easy for a human being to approximately find the centroid of a polygon but that a computer needs a lots of commands to achieve the same result.